src/components/ha-bar-slider.ts

```diff
@@ -0,0 +1,410 @@
+          e,
+          this.orientation === "vertical"
+        );
+        this.value = this.percentageToValue(percentage);
+        const value = this.steppedValue(this.value);
+        fireEvent(this, "slider-moved", { value });
+      });
+      this._mc.on("panend", (e) => {
+        if (this.disabled) return;
+        this.controlled = false;
+        const percentage = getPercentageFromEvent(
+          e,
+          this.orientation === "vertical"
+        );
+        this.value = this.steppedValue(this.percentageToValue(percentage));
+        fireEvent(this, "slider-moved", { value: undefined });
+        fireEvent(this, "value-changed", { value: this.value });
+      });
+
+      this._mc.on("singletap", (e) => {
+        if (this.disabled) return;
+        const percentage = getPercentageFromEvent(
+          e,
+          this.orientation === "vertical"
+        );
+        this.value = this.steppedValue(this.percentageToValue(percentage));
+        fireEvent(this, "value-changed", { value: this.value });
+      });
+    }
+  }
+
+  private get _tenPercentStep() {
+    return Math.max(this.step, (this.max - this.min) / 10);
+  }
+
+  _handleKeyDown(e: KeyboardEvent) {
+    if (!A11Y_KEY_CODES.has(e.code)) return;
+    e.preventDefault();
+    switch (e.code) {
+      case "ArrowRight":
+      case "ArrowUp":
+        this.value = this.boundedValue((this.value ?? 0) + this.step);
+        break;
+      case "ArrowLeft":
+      case "ArrowDown":
+        this.value = this.boundedValue((this.value ?? 0) - this.step);
+        break;
+      case "PageUp":
+        this.value = this.steppedValue(
+          this.boundedValue((this.value ?? 0) + this._tenPercentStep)
+        );
+        break;
+      case "PageDown":
+        this.value = this.steppedValue(
+          this.boundedValue((this.value ?? 0) - this._tenPercentStep)
+        );
+        break;
+      case "Home":
+        this.value = this.min;
+        break;
+      case "End":
+        this.value = this.max;
+        break;
+    }
+    fireEvent(this, "slider-moved", { value: this.value });
+  }
+
+  _handleKeyUp(e: KeyboardEvent) {
+    if (!A11Y_KEY_CODES.has(e.code)) return;
+    e.preventDefault();
+    fireEvent(this, "value-changed", { value: this.value });
+  }
+
+  destroyListeners() {
+    if (this._mc) {
+      this._mc.destroy();
+      this._mc = undefined;
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <div
+        id="slider"
+        class=${classMap({
+          slider: true,
+          controlled: this.controlled,
+        })}
+        style=${styleMap({
+          "--value": `${this.valueToPercentage(this.value ?? 0)}`,
+        })}
+        tabindex="0"
+        @keydown=${this._handleKeyDown}
+        @keyup=${this._handleKeyUp}
+        role="slider"
+        aria-valuemin=${this.min}
+        aria-valuemax=${this.max}
+        aria-valuenow=${this.value ?? 0}
+        aria-labelledby=${ifDefined(this.label)}
```

Comment:
`aria-labelledby` should point to the `id` of the element containing the visible label, otherwise you'd use `aria-label` if there isn't one.

My comment was that the component should be visibly rendering a label for the slider next to it.

Reply:
Added!

Reply:
You mean in the gallery?  The only element rendered by `ha-bar-slider` is the `div` with `role="slider".  It needs to render its own label from the `label` provided.

The value for `aria-labelledby` is currently invalid.  It takes the `id` of another element which has the label.

Reply:
To be clearer, something like:

```
<div id="label">${this.label}</div>
<div role="slider" aria-labelledby="label" ...>...</div>
```

And you'd probably want to add an icon option as well to possibly render next to or in place of the label text, but then it should have its own label and tooltip.

Reply:
Oops, I misunderstood the ˋaria-labelled-by`. I will fix that for the gallery.

The idea is to have vertical slider for the more-info dialog to adjust brightness, cover position, etc...
We will also use a smaller horizontal slider on lovelace cards. I don't think we will add icon or visible label next to the slider. 

Whatever solution we choose, accessibility will be taken into account.

Reply:
If an input control doesn't have a label or other instructions, it wouldn't be considered accessible (see [WCAG criterion 3.3.2](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)).

For smaller varieties on cards, an icon might suffice, or if that space really cannot be spared, then a tooltip at a minimum .

For detailed views like a more info dialog or dedicated card, there's no reason not to provide a text label with extra context for the min and max ends (e.g. "Brightness" with a bright light bulb at the high end and a dim one at the low end).

Reply:
`aria-label` has been added to vertical slider and `aria-labelledby` to horizontal slider. We have now the two example in the gallery 🙂
In Home Assistant Front-end labels will be rendered outside this slider with a `aria-labelledby` property but it's not part of this PR.